File: Leetcode - Top Interview Questions/11. Container With Most Water.cpp
class Solution
{
public:
    int maxArea(vector<int> &height)
    {
        int left(0), right(height.size() - 1), maxWater(0);
        while (left < right)
        {
            maxWater = max(maxWater, (right - left) * min(height[left], height[right]));
            if (height[left] > height[right])
                right--;
            else
                left++;
        }
        return maxWater;
    }
};

File: Leetcode - Top Interview Questions/11.Container_with_most_water.java
class Solution 
{
    public int maxArea(int[] height)
    {
        int area = Integer.MIN_VALUE;
        int start = 0, end = height.length-1;
        
        while(start < end)
        {
            area = Math.max(area, Math.min(height[start], height[end]) * (end - start));
            
            if(height[start] < height[end])
            {
                start += 1;
            } 
            else 
            {
                end -= 1;
            }
        }
        return area;
    }
}
